src/cares_wrap.cc

```diff
@@ -69,6 +69,17 @@ using v8::Value;
 
 namespace {
 
+#define CARES_16BIT(p) ((uint16_t)((unsigned int) 0xffff & \
+                         (((unsigned int)((unsigned char)(p)[0]) << 8U) | \
+                          ((unsigned int)((unsigned char)(p)[1])))))
+#define CARES_32BIT(p) ((unsigned int) \
+                         (((unsigned int)((unsigned char)(p)[0]) << 24U) | \
+                          ((unsigned int)((unsigned char)(p)[1]) << 16U) | \
+                          ((unsigned int)((unsigned char)(p)[2]) <<  8U) | \
+                          ((unsigned int)((unsigned char)(p)[3]))))
```

Comment:
Can you use inline functions rather than macros, and static casts rather than C-style casts?